src/lib/parallel/parallel.ml

```diff
@@ -17,3 +17,13 @@ let init_master () =
       ~rpc_handshake_timeout:(Time.Span.of_min 10.) ~rpc_heartbeat_config
       ~worker_command_args:[worker_command_name] () ;
     initialized := true )
+
+let kill ~logger ~prompt error =
+  let message = sprintf !"%s: %s!" prompt (Error.to_string_hum error) in
+  Logger.fatal logger !"%s!" message ;
+  failwithf !"%s!" message ()
+
+let run_connection ~logger ~error_message ~on_success action =
```

Comment:
Can you change this to use deferreds instead of callbacks? Tracing control flow through callbacks is usually challenging